activesupport/lib/active_support/core_ext/date_time/calculations.rb

```diff
@@ -43,12 +43,15 @@ def subsec
   # passed, then minute and sec is set to 0. If the hour and minute is passed,
   # then sec is set to 0. The +options+ parameter takes a hash with any of these
   # keys: <tt>:year</tt>, <tt>:month</tt>, <tt>:day</tt>, <tt>:hour</tt>,
-  # <tt>:min</tt>, <tt>:sec</tt>, <tt>:offset</tt>, <tt>:start</tt>.
+  # <tt>:min</tt>, <tt>:sec</tt>, <tt>:nsec</tt>, <tt>:usec</tt>,
+  # <tt>:offset</tt>, <tt>:start</tt>.
   #
   #   DateTime.new(2012, 8, 29, 22, 35, 0).change(day: 1)              # => DateTime.new(2012, 8, 1, 22, 35, 0)
   #   DateTime.new(2012, 8, 29, 22, 35, 0).change(year: 1981, day: 1)  # => DateTime.new(1981, 8, 1, 22, 35, 0)
   #   DateTime.new(2012, 8, 29, 22, 35, 0).change(year: 1981, hour: 0) # => DateTime.new(1981, 8, 29, 0, 0, 0)
   def change(options)
+    options.assert_valid_keys(:year, :month, :day, :hour, :min, :sec, :usec, :nsec, :offset, :start)
```

Comment:
Why don't we change this to take keyword arguments instead of doing the check ourselves?

Reply:
Good call. However I'll defer it to another PR, as such a change introduces subtle breaking behavior (due to `fetch` usage):

```ruby
# before
def change(options)
  new_year = options.fetch(:year, year)

# after
def change(year: nil, ...)
  new_year = year || self.year  # (slightly different behavior for falsy values)
```